packages/models/tests/unit/sql.spec.js

```diff
@@ -84,6 +84,31 @@ describe('normalize SQL', () => {
       normalize(`SELECT users.*, a.* FROM users WHERE a.user_id = 'alice'`)
     ).toEqual(`SELECT users.*, a.* FROM users WHERE a.user_id = ?`);
   });
+
+  test('removes comments correctly', () => {
+    expect(
+      normalize(
+        `SELECT 4 AS random; -- guaranteed random, generated by fair dice roll`
+      )
+    ).toEqual(`SELECT ? AS random; `);
+  });
+
+  test('handles comments in strings', () => {
+    expect(normalize(`SELECT 'hello -- world' AS title`)).toEqual(
+      `SELECT ? AS title`
+    );
+  });
+
+  // TODO: I would prefer if the whitespace was collapsed, but it's OK for now.
+  test('handles comments embedded in multiline SQL', () => {
+    expect(
+      normalize(`SELECT *
+-- this table contains users
+FROM users`)
+    ).toEqual(`SELECT *
+
+FROM users`);
```

Comment:
@dividedmind the way it leaves this whitespace isn't super ideal but it's OK for now. Just thought I'd point this out. I am not sure if normalizing whitespace is part of what this code is supposed to do anyway. Like you said, parse and emit by traversing the AST will be a better approach.

Reply:
Not really, it isn't, as far as I can tell. We're not normalizing spaces elsewhere either, I think.